app/services/sso/openid_connect.rb

```diff
@@ -45,6 +46,10 @@ def update_session(payload)
       session[:expires_at] = payload['exp']
     end
 
+    def http_token
+      request.env['HTTP_OIDC_ACCESS_TOKEN']
```

Comment:
@tbrisker 

> how is HTTP_OIDC_ACCESS_TOKEN header set? is this the default setting by keycloak sso or is it some header we define on the keycloak server?

The HTTP_OIDC_ACCESS_TOKEN is set by the `mod_auth_openidc` apache module. This is therefore a header that is inserted by the module and passed on to Foreman.

> In other words - is it possible to instead just set the authorization header like api requests and use the same flow?
Imo, that would not be the case.

Let me know your thoughts.

Reply:
Ok, thanks for the clarification.

Reply:
If that module is not present, any user can present it. It looks like the token is still validated regardless of how it came in, so that isn't a security concern, right?

> The HTTP_OIDC_ACCESS_TOKEN is set by the `mod_auth_openidc` apache module. This is therefore a header that is inserted by the module and passed on to Foreman.

That implies an installer change. Will this be required?

Reply:
@ekohl on second thought, should we also have a installer change that installs `mod_auth_openidc` for Foreman? some option like `--enable--mod_auth_openidc`.

What do you think?